site/en/integrations/integrate_with_sentencetransformers.md

```diff
@@ -6,222 +6,176 @@ title: Movie Search Using Milvus and SentenceTransformers
 
-## Inserting the data
+collection.create_index(
+    'embedding',
+    params
+)
+```
 
-For this example, we are going to use the SentenceTransformers miniLM model to create embeddings of the plot text. This model returns 384-dim embeddings.
+Once these steps are done, we are ready to insert data into the collection and perform a search. Any data added will be indexed automatically and be available to search immediately. If the data is very fresh, the search might be slower as brute force searching will be used on data that is still in process of getting indexed.
 
-In these next few steps we will be: 
+## Inserting the Data
 
-1. Loading the data.
-2. Embedding the plot text data using SentenceTransformers.
-3. Inserting the data into Milvus.
+For this example, we are going to use the SentenceTransformers miniLM model to create embeddings of the plot text. This model returns 384-dimension embeddings.
 
 ```python
-import csv
-from sentence_transformers import SentenceTransformer
-
-transformer = SentenceTransformer('all-MiniLM-L6-v2')
-
-# Extract the book titles
-def csv_load(file):
-    with open(file, newline='') as f:
-        reader = csv.reader(f, delimiter=',')
-        for row in reader:
-            if '' in (row[1], row[7]):
-                continue
-            yield (row[1], row[7])
-
-
-# Extract embedding from text using OpenAI
-def embed_insert(data):
-    embeds = transformer.encode(data[1]) 
-    ins = [
-            data[0],
-            [x for x in embeds]
-    ]
-    collection.insert(ins)
-
-import time
-
-data_batch = [[],[]]
+model = SentenceTransformer("all-MiniLM-L12-v2")
+```
 
-count = 0
+We loop over the rows of the data, embed the plot summary field, and insert entities into the vector database. In general, you should perform this step over batches of data items to maximize CPU or GPU throughput for the embedding model, as we do here.
 
-for title, plot in csv_load('./movies/plots.csv'):
-    if count <= COUNT:
-        data_batch[0].append(title)
-        data_batch[1].append(plot)
-        if len(data_batch[0]) % BATCH_SIZE == 0:
-            embed_insert(data_batch)
-            data_batch = [[],[]]
-        count += 1
-    else:
-        break
+```python
+for batch in tqdm(ds.batch(batch_size=512)):
+    embeddings = model.encode(batch['PlotSummary'])
+    data = [{"title": title, "embedding": embedding} for title, embedding in zip(batch['Title'], embeddings)]
+    res = collection.insert(data=data)
+```
 
-# Embed and insert the remainder
-if len(data_batch[0]) != 0:
-    embed_insert(data_batch)
+To be safe, we flush the data writing queue and check that the expected number of elements are present in the database.
```

Comment:
this isn't necessary, and it hurts performance. old milvus version used to requrie manual flush and it created a lot of confusion. we don't recommend manual flush anymore. see here for e2e code example: https://milvus.io/docs/insert-update-delete.md#Preparations